### PR TITLE
add some skeleton open api specs with bare bones structure and data m…

### PR DESCRIPTION
**Pull request by macreman**
…odels to be refined shortly

### CODE REVIEW

File: specs/EmergencyAuthCodeAPI.yml [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+          description: Bad request
+        '401':
+          description: Unauthorised
+components:
+  schemas:
+    companyOfficer:
+      type: object
+      required:
+        - id
+        - forename
```
Comment (jphillips2121) [inline specs/EmergencyAuthCodeAPI.yml:124]: I've just been playing around with returning CHIPS information and it seems like the officers have a forename1 and forename2 on the officer_details table. Should both be captured here as well?

Comment (macreman) [reply specs/EmergencyAuthCodeAPI.yml:124]: Possibly, I've gone with very basic placeholder data models at this point as I haven't had time to get into the required data model requirements in oracle and we don't have much in terms of UX at the moment so holding back on adding too much detail that we may need to either rework or remove later on e.g. service addresses, dob etc have been left out for now until we know if they will be required for the UX side of the service. Other parts have purposely been left out for now like the actual URA that we won't expose via the API but will be used behind the scenes.

**Comment by emead** [comment]
Are all the endpoints strictly necessary for a first release?
Depending on how the user journey is to be implemented (i.e. the screen order), could we potentially just implement:

1. GET list of officers
2. POST create authcode request

**Comment by macreman** [comment]
@emead possibly, it will depend on UX requirements to make the final decision on whats required in these specs so trying to cover all the bases

File: specs/EmergencyAuthCodeAPI.yml [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+          description: The officers surname
+          example: Bloggs
+    companyOfficers:
+      type: object
+      required:
+        - items
+        - total_count
+      properties:
+        items:
+          $ref: '#/components/schemas/companyOfficer'
```
Comment (jphillips2121) [inline specs/EmergencyAuthCodeAPI.yml:147]: When returning multiple officers should this be an array?

Comment (macreman) [reply specs/EmergencyAuthCodeAPI.yml]: nice catch, updated to an array and tweaked the surround count field structure to use standard pagination fields

File: specs/EmergencyAuthCodeAPI.yml (hunk 1 above)
Comment (macreman) [reply specs/EmergencyAuthCodeAPI.yml]: After seeing the prototype requirements I've added extra data to the officer models and simplified the "officer name" to a single field after cross referencing against our output data specs which combine all name elements into a single field already - these specs do sometimes offer a breakdown of name elements such as title, forename, other forename, surname, honours, etc. which can be added here if required later on but given the prototype screens as they are I don't see the need for this breakdown at this point

**Pull request merged**